compiler/rustc_target/src/spec/mod.rs

```diff
@@ -1925,6 +1925,24 @@ impl Arch {
             Self::Other(name) => rustc_span::Symbol::intern(name),
         }
     }
+
+    pub fn supports_c_variadic_definitions(&self) -> bool {
+        use Arch::*;
+
+        match self {
+            // These targets just do not support c-variadic definitions.
+            Bpf | SpirV => false,
+
+            // We don't know if the target supports c-variadic definitions, but we don't want
+            // to needlesly restrict custom target.json configurations.
+            Other(_) => true,
```

Comment:
This is kind of up for debate, but a custom `target.json` is kind of "here be dragons" already, and if the new target gets proper compiler support a choice still has to be made here.